File: src/Reporting.Server/Constants/ApiRoutes.cs
namespace Reporting.Server.Constants
{
    public static class ApiRoutes
    {

        public static class V1
        {

            public static class Reporting
            {
                public const string GetReports = "api/v1/reporting/active-reports";
                public const string GetReportDetails = "api/v1/reporting/report/{key}";
                public const string GetReportParameters = "api/v1/reporting/report/{key}/parameters";
                public const string GetReportColumnDefinitions = "api/v1/reporting/report/{key}/columns";             
                public const string GetReportData = "api/v1/reporting/report/{key}/data";
                public const string DownloadReport = "api/v1/reporting/report/{key}/file";

            	public static class Admin
            	{             
                    public const string GetReports = "api/v1/admin/reporting/all-reports";
                    public const string GetReportDetails = "api/v1/admin/reporting/report/{key}";
                    public const string GetReportSourceDetails = "api/v1/admin/reporting/report-source/{id}";
                    public const string GetAvailableReportSourcesForNewReport = "api/v1/admin/reporting/report/unused-report-sources";
                    public const string NewReport = "api/v1/admin/reporting/report";
                    public const string UpdateReport = "api/v1/admin/reporting/report";
                    public const string DeleteReport = "api/v1/admin/reporting/report";
            	}
            }
        }
    }
}

File: src/reporting-app-next/src/constants/api-routes.ts
const getUserRoute = (endpoint: string): string => {
    return `https://localhost:7260/api/v1/${endpoint}`;
};

const getAdminRoute = (endpoint: string): string => {
    return `https://localhost:7260/api/v1/${endpoint}`;
};

const UserApiRoutes = {
    GetReports: getUserRoute("reporting/active-reports"),
    GetReportDetails: (key: string) => getUserRoute(`reporting/report/${key}`),
    GetReportParameters: (key: string) => getUserRoute(`reporting/report/${key}/parameters`),
    GetReportColumnDefinitions: (key: string) => getUserRoute(`reporting/report/${key}/columns`),    
    GetReportData: (key: string) => getUserRoute(`reporting/report/${key}/data`),
    DownloadReport: (key: string) => getUserRoute(`reporting/report/${key}/file`),
};

const AdminApiRoutes = {
    GetReports: getAdminRoute("admin/reporting/all-reports"),
    GetReportDetails: (key: string) => getAdminRoute(`admin/reporting/report/${key}`),
    GetReportSourceDetails: (id: number) => getAdminRoute(`admin/reporting/report-source/${id}`),
    GetAvailableReportSourcesForNewReport: getAdminRoute("admin/reporting/report/unused-report-sources"),
    NewReport: getAdminRoute("admin/reporting/report"),
    UpdateReport: getAdminRoute("admin/reporting/report"),
    DeleteReport: getAdminRoute("admin/reporting/report"),
};

export { UserApiRoutes, AdminApiRoutes };
  